docs/SMT.md

```diff
@@ -378,10 +378,22 @@ This backstepping process allows the traversal to continue until it reaches a
 sentinel leaf that has the longest common prefix and most bits in common with
 the provided hash, up to the depth of the leaf found.
 
-This method guarentees a proof of inclusion in all cases and can be verified by
+This method guarantees a proof of inclusion in all cases and can be verified by
 using the `VerifyClosestProof` function which requires the proof and root hash
 of the trie.
 
+NB: If the hash provided to the `ClosestProof` function is known prior to the
+tree being filled and closed there is the possibility of placing a leaf where
+the hash will lead. If used **as intended**, the hash provided should **not** be
```

Comment:
I still am not clear on "who" we should be concerned about knowing the hash and "whom" would be placing a leaf where the hash will lead and whether they're the same actor.